app/controllers/wiki_pages_api_controller.rb

```diff
@@ -616,7 +616,9 @@ def get_update_params(allowed_fields=Set[])
     # needs to be moved to the controller where we know who is
     # editing it.
     if page_params.include?(:body)
-      unless BeyondZConfiguration.unrestricted_html_users.include?(@current_user.id)
+      # unless BeyondZConfiguration.unrestricted_html_users.include?(@current_user.id)
+      # This checks if the current user is a teacher of the course this page is attached to - replaces the old security whitelist.
+      unless @current_user && @context && @context.respond_to?(:grants_any_right?) && can_do(@context, @current_user, :manage)
```

Comment:
Wow, that is a nutty way to check roles.  It's amazing that all these "does this object exist" things aren't abstracted away in a helper or something.  Anyway, I think this is how I've seen it in the codebase.